src/index.js

```diff
@@ -115,17 +115,38 @@ function *selectRole(samlAssertion, roleName) {
   return role;
 }
 
+function parseRoleAttributeValue(attributeValue) {
+  let arns = attributeValue.split(',').map(function (arn) {
+    let arnParts = arn.split('/', 2);
+    let key = arnParts[0], value = arnParts[1];
+    let type = key.split('::', 2)[1].split(':', 2)[1];
```

Comment:
This code could be refined to split out the account number as well with something like `/arn:aws:iam::(\d+):(\w+)/(\w+)/` where `$1` is the account number, `$2` is the ARN type, and `$3` is the ARN value. The account number could be included in the return value from `parseRoleAttributeValue` below to allow for more consistent account handling as mentioned in my PR description.